src/Reactor.jl

```diff
@@ -9,11 +9,13 @@ using SparseArrays
 abstract type AbstractReactor end
 export AbstractReactor
 
-struct Reactor{D,Q} <: AbstractReactor
+struct Reactor{D,Q,F1,F2} <: AbstractReactor
     domain::D
     ode::ODEProblem
     recommendedsolver::Q
     forwardsensitivities::Bool
+    prec::F1 #function to calculate preconditioner
+    psetup::F2 #function to compute preconditioner \ residue
```

Comment:
Can we add all the functions necessary to do the preconditioner for the julia solvers?